php/libraries/Database.class.inc

```diff
@@ -614,13 +614,13 @@ class Database
      * @param array  $result an array into which the results of the select will
      *                       be stored - this array gets overwritten completely
      *
-     * @return     none
+     * @return     void
      * @access     public
      * @deprecated Use pselect wrappers instead
      */
     function selectRow($query, &$result)
     {
-        $this->select($query, $result);
+        $result = $this->pselect($query, array());
```

Comment:
shouldn't this function be deprecated ?

is this changes related to phan fixes ?

Reply:
same applies below

Reply:
yeah, it should be.. but do you think 20.0 is the release to do it? It's been deprecated for a while now, so maybe it's finally time, but this change was just to fix an error from phan (I can't remember which)

Reply:
I think it should either be completely removed or ignored by phan but I dont think we should update these functions to avoid people using them in the future...

Reply:
Oh, now I remember: it was to fix the error:

"Call to deprecated function \Database::select() defined at php/libraries/Database.class.inc:568"

so that PhanDeprecatedFunction could be removed from the ignore list.

Reply:
then I guess the solution would be to remove those

Reply:
I'll send a PR after this to remove, but I don't think it should be done as part of this PR, and I'm not convinced that we're ready with all the legacy instruments that might still be calling it. It's going to be really painful for projects to upgrade, but at least with this change we can be sure we're not calling anything that's been deprecated in the main codebase

Reply:
Arguably with this change they have no more reason to stop calling these functions. I think it's been over 3 years these functions are deprecated.

We can do a very quick survey see the effect on projects